orion-ui/src/compositions/useToast.ts

```diff
@@ -0,0 +1,19 @@
+import { showToast, Toast, ToastOptions, ToastType } from '@prefecthq/prefect-design'
+import { onUnmounted, Component } from 'vue'
+
+/**
+ * The useToast composition takes the same arguments as showToast but also dismisses the toast when the caller component is unmounted
```

Comment:
This is interesting in clever. But I think we could make `dismissOnUnmount` or something similar part of `ToastOptions` right? Rather than using a composition.

Reply:
Ultimately we might want to do that but @stackoverfloweth and I explored this a little and it would require a non-trivial refactor of the toast plugin